pkg/clients/ilmt/ilmt.go

```diff
@@ -56,110 +68,231 @@ func NewClient(config *IlmtConfig) Client {
+		var standaloneProductRespObj StandaloneProductResp
+		json.Unmarshal(standaloneProductRespData, &standaloneProductRespObj)
 
-	fmt.Println("Standalone Product Name List: ")
-	for i := 0; i < len(responseObject.ProductLicenceUsage); i++ {
-		fmt.Println(responseObject.ProductLicenceUsage[i].ProductName)
-	}
+		// licence usage for product that are part of bundle
+		productPartOfBndlResp, err := ilmtC.Do(urlForProductPartOfBndlUsage)
+		if err != nil {
+			log.Fatal(err.Error())
+			return -1, EMPTY, err
+		}
 
-	// licence usage for product that are part of bundle
-	prodPartOfBndlResp, err := ilmtC.Do(urlForProductPartOfBndlUsage)
-	if err != nil {
-		log.Fatal(err.Error())
-		return -1, EMPTY, err
-	}
+		productPartOfBndlRespData, err := ioutil.ReadAll(productPartOfBndlResp.Body)
+		if err != nil {
+			log.Fatal(err.Error())
+			return -1, EMPTY, err
+		}
 
-	prodPartOfBndlRespData, err := ioutil.ReadAll(prodPartOfBndlResp.Body)
-	if err != nil {
-		log.Fatal(err.Error())
-		return -1, EMPTY, err
-	}
-	// fmt.Println(string(prodPartOfBndlRespData))
+		var productPartOfBndlRespObj ProductPartOfBndlResp
+		json.Unmarshal(productPartOfBndlRespData, &productPartOfBndlRespObj)
 
-	var respObjPartOfBndl ProdPartOfBndlResp
-	json.Unmarshal(prodPartOfBndlRespData, &respObjPartOfBndl)
+		// licence usage for product that are part of bundlE
+		parentProductResp, err := ilmtC.Do(urlForParentProductUsage)
+		if err != nil {
+			log.Fatal(err.Error())
+			return -1, EMPTY, err
+		}
 
-	fmt.Println("")
-	fmt.Println("===============================================================================")
-	fmt.Println("Count of total no of products part of bundle for usage: ", respObjPartOfBndl.TotalRows)
-	fmt.Println("Count of products part of bundle usage fetched: ", len(respObjPartOfBndl.ProdPartOfBundlLicenceUsage))
+		parentProductRespData, err := ioutil.ReadAll(parentProductResp.Body)
+		if err != nil {
+			log.Fatal(err.Error())
+			return -1, EMPTY, err
+		}
 
-	fmt.Println("Product Part Of bundle Name List: ")
-	for i := 0; i < len(respObjPartOfBndl.ProdPartOfBundlLicenceUsage); i++ {
-		fmt.Println(respObjPartOfBndl.ProdPartOfBundlLicenceUsage[i].ProductName)
-	}
+		var parentProductRespObj ParentProductResp
+		json.Unmarshal(parentProductRespData, &parentProductRespObj)
 
-	// licence usage for product that are part of bundlE
-	parentProdResp, err := ilmtC.Do(urlForParentProductUsage)
+		productUsageCount := len(standaloneProductRespObj.StandaloneProductLicenceUsage) + len(productPartOfBndlRespObj.ProductPartOfBndlLicenceUsage)
+		productUsageTrnsfrmdEvntDataSlice := make([]ProductUsageTransformedEventData, productUsageCount)
 
-	if err != nil {
-		log.Fatal(err.Error())
-		return -1, EMPTY, err
-	}
+		counter := 0
+		productUsageTransformedEventJson := []byte{}
+
+		for _, productResp := range standaloneProductRespObj.StandaloneProductLicenceUsage {
+
+			startDateMillis := startDate.UnixMilli()
```

Comment:
Can we modularize `FetchUsageData` method ? - by moving the transformation logic into a separate method?

Reply:
Dear @MJSiddu , This has been developed as per the ILMT usage documentation and the discussion which we had Lan, Gautam, Zach. That is why there is no prompt for the end date and it picks up the yesterday date as a default. FYI @dacleyra, Gautam Zalpuri [gzalpuri@us.ibm.com](mailto:gzalpuri@us.ibm.com), Lan Vuong [lvuong@us.ibm.com](mailto:lvuong@us.ibm.com)

Reply:
Dear @MJSiddu yes we will modularize the transformation logic in to separate method.

Reply:
Yes, everything from fetch to Tar will be a single process, but that doesn't mean we put all the code in 1 single method.  This method is growing, and it's already more than 400 lines of code inside 1 method. So I'm just asking you to modularize it (without changing the business logic/flow)